cmd/runtests/runner.go

```diff
@@ -58,9 +61,58 @@ func (r *runner) run(ctx context.Context, cmd *cobra.Command, args []string) err
 
 	// test client and authentication
 	err := uat.TestClient(giantSwarmClient, authWriter)
-	if err != nil {
-		return microerror.Mask(err)
+	cliutil.ExitIfError(err)
+
+	// 1. Create a cluster with one node pool based on defaults.
+	fmt.Printf("\nStep 1 - Create a cluster with one node pool based on defaults\n")
+	clusterOneID, clusterOneAPIEndpoint, err := uat.Test01ClusterCreation(giantSwarmClient, authWriter)
+	cliutil.ExitIfError(err)
+
+	time.Sleep(3 * time.Second)
+
+	// Workaround until step 1 returns proper cluster info.
+	if clusterOneAPIEndpoint == "" {
+		fmt.Printf("\nStep 1a - Get cluster details, so we know the API endpoint\n")
+		var params *clusters.GetClusterV5Params
+		params = clusters.NewGetClusterV5Params().WithClusterID(clusterOneID)
+		result, err := giantSwarmClient.Clusters.GetClusterV5(params, authWriter)
+		if err != nil {
+			cliutil.ExitIfError(microerror.Mask(err))
+		}
+
+		clusterOneAPIEndpoint = result.Payload.APIEndpoint
+	}
+
+	time.Sleep(3 * time.Second)
```

Comment:
Why is the delay here still necessary? After we improved `cluster-service` with a backoff for the key pair creation I thought this is fine now.

Reply:
No, actually cluster details couldn't be fetched right after cluster creation either.